core/domain/services/registration/CopyRegistrationService.php

```diff
@@ -174,4 +176,45 @@ public function copyPaymentDetails(
         }
         return true;
     }
+
+
+    /**
+     * @param EE_Registration $target_registration
+     * @param EE_Registration $registration_to_copy
+     * @return bool
+     * @throws RuntimeException
+     * @throws UnexpectedEntityException
+     * @throws EE_Error
+     */
+    public function copyPromotionLineItems(
+        EE_Registration $target_registration,
+        EE_Registration $registration_to_copy
+    ) {
+        $transaction = $registration_to_copy->transaction();
+        if ($transaction instanceof \EE_Transaction) {
```

Comment:
plz import all classes so that all dependencies are listed at the top of the file
so in other words, this will change to 

```suggestion
        if ($transaction instanceof EE_Transaction) {
```

and then you'll add:

```php
use EE_Transaction;
```

at the top with the other imports.

Then do this for all other classes still using the root namespace `\`